packages/tools/devtools/devtools-view/src/components/data-visualization/EditableView.tsx

```diff
@@ -0,0 +1,364 @@
+					node={node}
+					setEditProp={setEditProp}
+					editProp={editProp}
+					submitChange={submitChange}
+				/>
+			);
+			break;
+		case "undefined":
+			innerView = <i>: undefined</i>;
+			break;
+
+		case "null":
+			innerView = <i>: null</i>;
+			break;
+
+		default:
+			throw new Error("Unrecognized edit type.");
+	}
+
+	const styles = useStyles();
+
+	return (
+		<div
+			className={editProp !== undefined ? styles.isEditingStyle : styles.isNotEditingStyle}
+			onBlur={onBlur}
+			onClick={(event): void => event?.preventDefault()}
+		>
+			<TreeHeader label={label} nodeTypeMetadata={node.typeMetadata} />
+			<Dropdown
+				size="small"
+				className={styles.dropdownStyle}
+				onOptionSelect={onOptionSelect}
+				value={
+					editProp === undefined
+						? node.value === null
+							? "null"
+							: typeof node.value
+						: editProp.type
+				}
+				selectedOptions={[
+					editProp === undefined
+						? node.value === null
+							? "null"
+							: typeof node.value
+						: editProp.type,
+				]}
+			>
+				{options.map((option) => (
+					<Option
+						onMouseDownCapture={(event): void => {
+							event.preventDefault();
+						}}
+						key={option}
+					>
+						{option}
+					</Option>
+				))}
+			</Dropdown>
+			{innerView}
+		</div>
+	);
+}
+
+/**
+ * Component which allows for boolean editing
+ */
+function EditableBooleanComponent(props: EditableComponentProps): React.ReactElement {
+	const { node, submitChange } = props;
+
+	/**
+	 * Auxiliary function to make boolean editing buttons
+	 * @param isTrue - Determines if it should be a true or false button
+	 * @returns 2 buttons, one for setting value to false and one for true
+	 */
+	function boolButton(isTrue: boolean): React.ReactElement {
+		return (
+			<Button
+				size="small"
+				disabled={isTrue === node.value}
+				onClick={(event): void => {
+					event.preventDefault();
+					event.stopPropagation();
+					submitChange(isTrue);
+				}}
+			>
+				{String(isTrue).charAt(0).toUpperCase() + String(isTrue).slice(1)}
+			</Button>
+		);
+	}
+	return (
+		<>
+			{boolButton(true)}
+			{boolButton(false)}
+		</>
+	);
+}
+
+/**
+ * Input for {@link EditableInputComponent}
+ */
+interface EditableInputComponent extends EditableComponentProps {
+	inputType: "string" | "number";
+}
+
+/**
+ * Component which allows for number editing
+ */
+function EditableInputComponent(props: EditableInputComponent): React.ReactElement {
+	const { node, submitChange, inputType, editProp, setEditProp } = props;
+
+	/**
+	 * Provides confirming and canceling an edit functionality
+	 */
+	const onKeyDown = React.useCallback(
+		(event: React.KeyboardEvent<HTMLInputElement>) => {
+			const key = event.key;
+
+			if (editProp === undefined) {
+				return;
+			}
+			if (key === "Enter") {
+				submitChange(editProp.type === "number" ? Number(editProp.value) : editProp.value);
+				event.currentTarget.blur();
+			}
+
+			if (key === "Escape") {
+				event.currentTarget.blur();
```

Comment:
I think you can replace this with `setEditProp(undefined)` - since that also cancels editing mode?